src/pudl_rmi/create_override_spreadsheets.py

```diff
@@ -237,13 +237,10 @@ def _prep_deprish(deprish, pudl_out) -> pd.DataFrame:
     logger.debug("Prepping Deprish Data")
 
     # Get utility_id_eia from EIA
-    util_df = pudl_out.utils_eia860()
-    id_dict = dict(zip(util_df["utility_id_pudl"], util_df["utility_id_eia"]))
-
-    deprish_out = deprish.assign(
-        report_year=lambda x: x.report_date.dt.year.astype("Int64"),
-        utility_id_eia=lambda x: x.utility_id_pudl.map(id_dict).astype("Int64"),
-    )
+    util_df = pudl_out.utils_eia860()[["utility_id_pudl", "utility_id_eia"]]
+    deprish_out = deprish.copy()
```

Comment:
i think this copy is actually not necessary. because you are not using and then editing a slice of the og df. the `loc` and `merge` below are both operating on the full df (well the `loc` is operating on a slice but is the sanctioned way to edit a slice of a df w/o needing to copy it)

but overall this new function looks great!

Reply:
ok question on this -- my understanding is that if you don't use `.copy()` and you make _edits_ to the slice, then you are editing the original df. This particular function takes the deprish file from pudl_out, makes a new column called `report_year`, and merges that with the util_df. The reason I put a `.copy()` here is because I didn't want the util_df stored at `pudl_out.utils_eia860()` to now have a `report_year` column.

Reply:
I am going to be honest, I do not totally understand this... but for the `pudl_out` specifically, I have tested this a few times and it seems like adding columns to the original does not edit the underlying df:

<img width="639" alt="image" src="https://user-images.githubusercontent.com/25487998/166305824-3870d08a-f979-4941-b374-5bc6b85f208e.png">

In this case, the copy is being applied to `deprish`. unless you actually want to preserve the original i don't see the need for this. I could be wrong though the copy stuff is a little nebulous to me

Reply:
my bad. you are right here... assign must create a copy itself bc this does not produce an error
 ```
bf = pudl_out.bf_eia923()
bf['test_col'] = 1
pudl_out.bf_eia923().test_col
```

Reply:
verdict is no need for .copy()